src/uct/ib/dc/base/dc_ep.h

```diff
@@ -224,4 +244,36 @@ static inline ucs_status_t uct_dc_iface_dci_get_dcs(uct_dc_iface_t *iface, uct_d
             return status; \
         } \
     }
+
+
+/* First, check whether we have FC window. If hard threshold is reached, credit
+ * request will be sent by "fc_ctrl" as a separate message. TX resources
+ * are checked after FC, because fc credits request may consume latest
+ * available TX resources. */
+#define UCT_DC_CHECK_RES_AND_FC(_iface, _ep) \
+    { \
+        if (ucs_unlikely((_ep)->fc.fc_wnd <= \
+                         (_iface)->super.config.fc_hard_thresh)) { \
+             if ((_iface)->super.config.fc_enabled) { \
```

Comment:
can we put the code which comes after checking fc_wnd<hard_thresh into a function instead of a macro?
it would be slow-path anyway, and we could avoid long macros